data_validation/data_validation.py

```diff
@@ -81,8 +81,10 @@ def __enter__(self):
     def __exit__(self, exc_type, exc_value, exc_traceback):
         if hasattr(self, "config_manager"):
             try:
-                self.config_manager.source_client.con.dispose()
-                self.config_manager.target_client.con.dispose()
+                if hasattr(self.config_manager.source_client, "con"):
+                    self.config_manager.source_client.con.dispose()
```

Comment:
We might want to check if dispose() is a supported method too. I know in Teradata I get the following now: `PM-WARNING: Exception closing connections: 'TeradataCursor' object has no attribute 'dispose'`

Reply:
> We might want to check if dispose() is a supported method too. I know in Teradata I get the following now: `PM-WARNING: Exception closing connections: 'TeradataCursor' object has no attribute 'dispose'`

Hmm, I think I should wrap the `dispose()` calls in a check for Oracle and PostgreSQL. They are the systems with a known problem. I made the assumption that there would be consistency across other engine's interfaces but that was clearly a bad assumption.